back/app/models/initiative.rb

```diff
@@ -96,9 +96,10 @@ class Initiative < ApplicationRecord
   end)
 
   scope :with_status_code, (proc do |code|
-    joins('LEFT OUTER JOIN initiative_initiative_statuses ON initiatives.id = initiative_initiative_statuses.initiative_id')
-      .joins('LEFT OUTER JOIN initiative_statuses ON initiative_statuses.id = initiative_initiative_statuses.initiative_status_id')
-      .where(initiative_statuses: { code: code })
+    left_outer_joins(:initiative_initiative_status)
+      .where(initiative_initiative_statuses: {
+        initiative_status_id: InitiativeStatus.where(code: code).select(:id)
+      })
```

Comment:
I replaced the code with that I originally developed for the `InitiatitivePolicy` scope, because using the original model scope in the policy leads to the following errors in the [filter_counts endpoint acceptance tests](https://github.com/CitizenLabDotCo/citizenlab/blob/3ee280aa962f47601a99200f622c6452cf5577cd/back/spec/acceptance/initiatives_spec.rb#L259-L281):
```
# PG::DuplicateAlias:
#   ERROR:  table name "initiative_initiative_statuses" specified more than once
#   /bundle/ruby/2.7.0/gems/rack-mini-profiler-3.1.0/lib/patches/db/pg.rb:69:in `exec_params'
```
Probably because the [query used in the filter_counts action](https://github.com/CitizenLabDotCo/citizenlab/blob/3ee280aa962f47601a99200f622c6452cf5577cd/back/app/controllers/web_api/v1/initiatives_controller.rb#L63-L75) also specifies the same table name.

Let's discuss this on Monday, as I'd like to understand more about this error, and also which approach is preferable (this, or my previous approach, or probably some other approach).